event.go

```diff
@@ -0,0 +1,376 @@
+type ChannelUnhold struct {
+	EventData
+	ChannelData
+}
+
+// ChannelUserevent events are custom-formatted events which have been received
+//   TODO: figure out if these include AMI user events, etc, and how the data is
+//   formatted
+type ChannelUserevent struct {
+	EventData
+	Bridge Bridge `json:"bridge,omitempty"`
+	ChannelData
+	Endpoint  Endpoint    `json:"endpoint,omitempty"`
+	Eventname string      `json:"eventname"` // Name of the user event
+	Userevent interface{} `json:"userevent"` // Custome data sent with the user event
+}
+
+// ChannelVarset events indicate a channel variable has been set (or changed)
+type ChannelVarset struct {
+	EventData
+	ChannelData
+	Value    string `json:"value"`    // New value
+	Variable string `json:"variable"` // Variable name
+}
+
+/* --------- Contact Events ---------------- */
+
+// ContactStatusChanged events indicate that a contact state has changed
+type ContactStatusChanged struct {
+	EventData
+	Endpoint    Endpoint    `json:"endpoint"`
+	ContactInfo ContactInfo `json:"contact_info"`
+}
+
+// DeviceStateChanged events indicate that a device state has changed
+type DeviceStateChanged struct {
+	EventData
+	Device_state DeviceState `json:"device_state"`
+}
+
+// Dial events indicate the dialing state for a channel has changed
+type Dial struct {
+	EventData
+	Caller     Channel `json:"caller,omitempty"`     // Dialing channel (if not system-originated)
+	Dialstatus string  `json:"dialstatus"`           // Present status of the dial attempt
+	Dialstring string  `json:"dialstring,omitempty"` // The string describing the dial
+	Forward    string  `json:"forward,omitempty"`    // If present, indicates the forwarding target TODO: what is the format of this?
+	Forwarded  Channel `json:"forwarded,omitempty"`  // If present, channel of forwarding target
+	Peer       Channel `json:"peer"`                 // Dialed channel
+}
+
+// EndpointStateChange events indicate that an endpoint has changed state
+type EndpointStateChange struct {
+	EventData
+	Endpoint Endpoint `json:"endpoint"`
+}
+
+/* --------- Peer Events ---------------- */
+
+// PeerStatusChange events indicate that a peer has changed state
+type PeerStatusChange struct {
+	EventData
+	Endpoint Endpoint `json:"endpoint"`
+	Peer     Peer     `json:"peer"`
+}
+
+/* --------- Playback Events ---------------- */
+
+// PlaybackFinished events indicate that a media playback operation has completed
+type PlaybackFinished struct {
+	EventData
+	Playback Playback `json:"playback"`
+}
+
+// PlaybackStarted events indicate that a media playback operation has begun
+type PlaybackStarted struct {
+	EventData
+	Playback Playback `json:"playback"`
+}
+
+/* --------- Recording Events ---------------- */
+
+/* These three are commented out until LiveRecording is taken care of
+// RecordingFailed events indicate that a recording operation request has failed to complete
+type RecordingFailed struct {
+	EventData
+	Recording LiveRecording `json:"recording"`
```

Comment:
Fixed in https://github.com/CyCoreSystems/ari/pull/28/commits/6067360c2151c6dfad49c214e58a3bd54b95977d